cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/BasicWorkflowContext.java

```diff
@@ -35,16 +47,46 @@
   private final ProgramWorkflowRunner programWorkflowRunner;
   private final Map<String, String> runtimeArgs;
   private final WorkflowToken token;
+  private final Metrics userMetrics;
 
   BasicWorkflowContext(WorkflowSpecification workflowSpec, @Nullable WorkflowActionSpecification specification,
                        long logicalStartTime, @Nullable ProgramWorkflowRunner programWorkflowRunner,
                        Map<String, String> runtimeArgs, WorkflowToken token) {
+    this(workflowSpec, specification, logicalStartTime, programWorkflowRunner,
+         ImmutableMap.copyOf(runtimeArgs), token, null, null, null, null, null, null);
+  }
```

Comment:
We may not need this constructor now. Always create `BasicWorkflowContext` with all values of the parameters.